modules/standard/IO.chpl

```diff
@@ -4161,66 +4225,73 @@ proc channel.writeln(const args ...?k,
   accessing this file concurrently.
   Unlike :proc:`file.fsync`, this does not commit the written data
   to the file's device.
-
-  :arg error: optional argument to capture an error code. If this argument
-              is not provided and an error is encountered, this function
-              will halt with an error message.
-
 */
-proc channel.flush(out error:syserr) {
-  error = ENOERR;
+proc channel.flush() throws {
+  var err:syserr = ENOERR;
   on this.home {
-    error = qio_channel_flush(locking, _channel_internal);
+    err = qio_channel_flush(locking, _channel_internal);
   }
+  if err then try this._ch_ioerror(err, "in channel.flush");
 }
 // documented in error= version
 pragma "no doc"
-proc channel.flush() throws {
-  var e:syserr = ENOERR;
-  this.flush(error=e);
-  if e then try this._ch_ioerror(e, "in channel.flush");
+proc channel.flush(out error:syserr) {
+  error = ENOERR;
+  try {
+    this.flush();
+  } catch e: SystemError {
+    error = e.err;
+  } catch {
+    error = EINVAL;
+  }
 }
 
-/* Assert that a channel has reached end-of-file.
-   Throws an error if the receiving channel is not currently
-   at EOF.
-
-   :arg error: an optional string argument which will be printed
-               out if the assert fails. The default prints "Not at EOF".
+/* Returns true if a channel has reached end-of-file, false if not.
+   Throws an error if this is a writing channel, or if there was
+   an error doing the read.
  */
-proc channel.assertEOF(error:string = "- Not at EOF") throws {
+proc channel.atEOF(): bool throws {
```

Comment:
It was intentional that we didn't have an `atEOF` function because such a function is inherently racy for channels that can get more data (e.g. a socket). Can this be a no-doc'd function that we just use to implement assertEOF?